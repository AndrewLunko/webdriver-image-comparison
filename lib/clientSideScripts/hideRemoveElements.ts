/**
 * Hide or remove elements on the page
 */
export default function hideRemoveElements(hideRemoveElements: { hide: HTMLElement[], remove: HTMLElement[] }, hideRemove: boolean): void {
  hideRemoveElements.hide.forEach(element => element.style.visibility = hideRemove ? 'hidden' : '');
  hideRemoveElements.remove.forEach(element => element.style.display = hideRemove ? 'none' : '');
}
