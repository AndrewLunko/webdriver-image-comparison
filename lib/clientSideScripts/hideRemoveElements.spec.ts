import hideRemoveElements from './hideRemoveElements';

describe('hideRemoveElements', () => {
  it('should be able to hide elements and put them back again', () => {
    document.body.innerHTML =
      '<div>' +
      '   <span id="id-1">Hello</span>' +
      '   <span id="id-2">Hello</span>' +
      '   <div>' +
      '     <span id="id-3">Hello</span>' +
      '     <span id="id-4">Hello</span>' +
      '  </div>' +
      '</div>';

    // Check not hidden
    expect((<HTMLElement>document.querySelector('#id-1')).style.visibility).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-3')).style.visibility).toMatchSnapshot();

    hideRemoveElements({
        hide: [document.querySelector('#id-1'), document.querySelector('#id-3')],
        remove: [],
      },
      true,
    );

    // Check hidden
    expect((<HTMLElement>document.querySelector('#id-1')).style.visibility).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-3')).style.visibility).toMatchSnapshot();

    hideRemoveElements({
        hide: [document.querySelector('#id-1'), document.querySelector('#id-3')],
        remove: [],
      },
      false,
    );

    // Check not hidden
    expect((<HTMLElement>document.querySelector('#id-1')).style.visibility).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-3')).style.visibility).toMatchSnapshot();
  });

  it('should be able to remove elements and put them back again', () => {
    document.body.innerHTML =
      '<div>' +
      '   <span id="id-1">Hello</span>' +
      '   <span id="id-2">Hello</span>' +
      '   <div>' +
      '     <span id="id-3">Hello</span>' +
      '     <span id="id-4">Hello</span>' +
      '  </div>' +
      '</div>';

    // Check not removed
    expect((<HTMLElement>document.querySelector('#id-2')).style.display).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-4')).style.display).toMatchSnapshot();

    hideRemoveElements({
        hide: [],
        remove: [document.querySelector('#id-2'), document.querySelector('#id-4')],
      },
      true,
    );

    // Check removed
    expect((<HTMLElement>document.querySelector('#id-2')).style.display).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-4')).style.display).toMatchSnapshot();

    hideRemoveElements({
        remove: [document.querySelector('#id-2'), document.querySelector('#id-4')],
        hide: [],
      },
      false,
    );

    // Check not removed
    expect((<HTMLElement>document.querySelector('#id-2')).style.display).toMatchSnapshot();
    expect((<HTMLElement>document.querySelector('#id-4')).style.display).toMatchSnapshot();
  });
});
