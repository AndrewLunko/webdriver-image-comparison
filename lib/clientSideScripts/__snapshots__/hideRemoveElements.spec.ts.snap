// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hideRemoveElements should be able to hide elements and put them back again 1`] = `""`;

exports[`hideRemoveElements should be able to hide elements and put them back again 2`] = `""`;

exports[`hideRemoveElements should be able to hide elements and put them back again 3`] = `"hidden"`;

exports[`hideRemoveElements should be able to hide elements and put them back again 4`] = `"hidden"`;

exports[`hideRemoveElements should be able to hide elements and put them back again 5`] = `""`;

exports[`hideRemoveElements should be able to hide elements and put them back again 6`] = `""`;

exports[`hideRemoveElements should be able to remove elements and put them back again 1`] = `""`;

exports[`hideRemoveElements should be able to remove elements and put them back again 2`] = `""`;

exports[`hideRemoveElements should be able to remove elements and put them back again 3`] = `"none"`;

exports[`hideRemoveElements should be able to remove elements and put them back again 4`] = `"none"`;

exports[`hideRemoveElements should be able to remove elements and put them back again 5`] = `""`;

exports[`hideRemoveElements should be able to remove elements and put them back again 6`] = `""`;
